pom.xml

```diff
@@ -1019,17 +1021,22 @@
         <module>client-flink/flink-1.14-shaded</module>
       </modules>
       <properties>
-        <flink.version>1.14.0</flink.version>
-        <flink.binary.version>1.14</flink.binary.version>
+        <flink.version>1.14.6</flink.version>
         <jackson.version>2.6.7</jackson.version>
         <jackson.databind.version>2.6.7.3</jackson.databind.version>
         <lz4-java.version>1.4.0</lz4-java.version>
-        <scala.version>2.12.15</scala.version>
-        <scala.binary.version>2.12</scala.binary.version>
         <zstd-jni.version>1.4.4-3</zstd-jni.version>
       </properties>
     </profile>
 
+    <profile>
+      <id>scala-2.11</id>
```

Comment:
shall we make scala version consist with flink version, like spark does?

Reply:
It seems  a bit redundant for those which need support two versions of scala.

Reply:
Technically, `spark-X.Y` and `flink-X.Y` can compose w/ `scala-X.Y`, e.g. `spark-3.3` can compose w/ `scala-2.12`, `scala-2.13`, on the other hand, it's rarely using no-default Scala version in Spark workloads.

Reply:
If Flink is in the same position, I recommend to drop `scala-X.Y` currently.

Reply:
Flink 1.14 officially supports scala 2.11 and 2.12, see [flink-1.14.6](https://archive.apache.org/dist/flink/flink-1.14.6/), and if the version higher than 1.14 will only supports 2.12+, so I think It would be appropriate to make 2.12 the default version while giving users the option to choose 2.11.

Reply:
> 

Users can specify scala version by -Dscala.version=xxx. IMO we don't need to add a profile for each scala version.

Reply:
> > 
> 
> Users can specify scala version by -Dscala.version=xxx. IMO we don't need to add a profile for each scala version.

Scala-2.11 profile is short for setting -Dscala.version=xxx -Dscala.binary.version=xxx and it's useful to keep binary and scala compile version consistent...

Reply:
Now we use scala-2.11 as default scala version for Flink.

Reply:
why scala-2.11 is chosen, it there a survey result/facts about the adoption of Flink 1.14 w/ Scala 2.11 or 2.12?

Reply:
It's the default version used in [Flink pom](https://github.com/apache/flink/blob/release-1.14.6/pom.xml).